### PR TITLE
Add an AdversarialExampleObjective: worst_with_constraint

### PR DESCRIPTION
**Pull request by jia-kai**
I find that in some cases using the `worst` as adversarial example objective results in much faster solving. However the current implementation does not add the constraint to enforce a successful attack, and the caller needs to check `ObjectiveValue` and `SolveStatus` to decide if the attack succeeds.

Therefore I added a new objective `worst_with_constraint` so the caller can observe the same behavior as of `closest`.

### CODE REVIEW

**Comment by vtjeng** [comment]
Thanks for the PR! I've been really busy these past few days, but will take a look at it as soon as I can.

**Comment by vtjeng** [comment]
Hi, sorry it took forever for me to review this. A lot has changed in this repo since you submitted the PR, but your PR correctly pointed to a bug in the code that I finally got around to understanding and fixing (see #152). I've also added you as a [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-github) to that PR so that you're correctly credited for your contributions (if I got it right). 

Thank you for your interest, and sorry again for taking so long to respond!

**Comment by jia-kai** [comment]
> **tl;dr**: Implemented in #152.
> 
> Hi, sorry it took forever for me to review this. A lot has changed in this repo since you submitted the PR, but your PR correctly pointed to a bug in the code that I finally got around to understanding and fixing (see #152). I've also added you as a [co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-github) to that PR so that you're correctly credited for your contributions (if I got it right).
> 
> Thank you for your interest, and sorry again for taking so long to respond!

Thank you! The new fix looks great.
